code/__DEFINES/callbacks.dm

```diff
@@ -1,6 +1,6 @@
 #define GLOBAL_PROC "some_magic_bullshit"
 /// A shorthand for the callback datum, [documented here](datum/callback.html)
 #define CALLBACK new /datum/callback
-#define INVOKE_ASYNC world.ImmediateInvokeAsync
+#define INVOKE_ASYNC(TRG, PR, ARG...) if ((TRG) == GLOBAL_PROC) { spawn (-1) call(PR)(##ARG); } else { spawn(-1) call(TRG, PR)(##ARG); }
```

Comment:
All these names need to be fully written out and snake_case

Also preferably this should be multilined because it's really hard to read